### ISSUE TITLE
ghcid doesn't work on modules with LINE pragmas

### ISSUE THREAD

**Issue by woehr** [ndmitchell/ghcid#275]
I've been working on a lexer and parser using Alex and Happy. The tools generate haskell files with LINE pragmas, but when the generated files are loaded, ghcid reports "All good" regardless of whether there are errors in the file or not. Test case below.

--------------

Given the following file, LinePragmaBad.hs:
```
{-# LINE 1 "LinePragmaBad.foo" #-}

module LinePragmaBad where

main :: IO ()
main = asdf
```

Running ghcid with the following produces an "All good" message despite there being an error in the file.

```
$ ghcid -c ghci LinePragmaBad
Loading ghci LinePragmaBad ...
GHCi, version 8.6.4: http://www.haskell.org/ghc/  :? for help
[1 of 1] Compiling LinePragmaBad    ( LinePragmaBad.hs, interpreted )

LinePragmaBad.foo:5:8: error: Variable not in scope: asdf :: IO ()
Failed, no modules loaded.

All good (1 module, at 16:46:12)
```

Verbose output:
```
$ ghcid -c ghci --verbose LinePragmaBad
%OS: linux
%ARCH: x86_64
%VERSION: 0.7.5
Loading ghci LinePragmaBad ...
%STDOUT: GHCi, version 8.6.4: http://www.haskell.org/ghc/  :? for help
%STDOUT2: GHCi, version 8.6.4: http://www.haskell.org/ghc/  :? for help
%STDIN: import qualified System.IO as INTERNAL_GHCID
%STDIN: :unset +t +s
%STDIN: :set prompt #~GHCID-START~#
%STDIN: :set -fno-break-on-exception
%STDIN: :set -fno-break-on-error
%STDIN: :set -v1
%STDIN: :set -fno-hide-source-paths
%STDIN: INTERNAL_GHCID.putStrLn ['#','~','G','H','C','I','D','-','F','I','N','I','S','H','-','1','~','#']
INTERNAL_GHCID.hPutStrLn INTERNAL_GHCID.stderr ['#','~','G','H','C','I','D','-','F','I','N','I','S','H','-','1','~','#']
GHCi, version 8.6.4: http://www.haskell.org/ghc/  :? for help
%STDOUT: [1 of 1] Compiling LinePragmaBad    ( LinePragmaBad.hs, interpreted )
%STDOUT2: [1 of 1] Compiling LinePragmaBad    ( LinePragmaBad.hs, interpreted )
[1 of 1] Compiling LinePragmaBad    ( LinePragmaBad.hs, interpreted )
%STDERR:
%STDOUT2:

%STDERR: LinePragmaBad.foo:5:8: error: Variable not in scope: asdf :: IO ()
%STDOUT2: LinePragmaBad.foo:5:8: error: Variable not in scope: asdf :: IO ()
LinePragmaBad.foo:5:8: error: Variable not in scope: asdf :: IO ()
%STDOUT: Failed, no modules loaded.
%STDOUT2: Failed, no modules loaded.
Failed, no modules loaded.
%STDOUT: Prelude> Prelude> Prelude INTERNAL_GHCID> Prelude INTERNAL_GHCID> #~GHCID-START~##~GHCID-START~##~GHCID-START~##~GHCID-START~##~GHCID-START~##~GHCID-FINISH-1~#
%STDERR: #~GHCID-FINISH-1~#
%STDIN:
%STDIN: INTERNAL_GHCID.putStrLn ['#','~','G','H','C','I','D','-','F','I','N','I','S','H','-','2','~','#']
INTERNAL_GHCID.hPutStrLn INTERNAL_GHCID.stderr ['#','~','G','H','C','I','D','-','F','I','N','I','S','H','-','2','~','#']
%STDOUT: #~GHCID-START~##~GHCID-START~##~GHCID-START~##~GHCID-FINISH-2~#
%STDERR: #~GHCID-FINISH-2~#
%STDIN: :set -ferror-spans
:set -j
:set -fdiagnostics-color=always

%STDIN: INTERNAL_GHCID.putStrLn ['#','~','G','H','C','I','D','-','F','I','N','I','S','H','-','3','~','#']
INTERNAL_GHCID.hPutStrLn INTERNAL_GHCID.stderr ['#','~','G','H','C','I','D','-','F','I','N','I','S','H','-','3','~','#']
%STDOUT: #~GHCID-START~##~GHCID-START~##~GHCID-START~##~GHCID-START~##~GHCID-START~##~GHCID-START~##~GHCID-FINISH-3~#
%STDERR: #~GHCID-FINISH-3~#
%STDIN: :show paths
%STDIN: INTERNAL_GHCID.putStrLn ['#','~','G','H','C','I','D','-','F','I','N','I','S','H','-','4','~','#']
INTERNAL_GHCID.hPutStrLn INTERNAL_GHCID.stderr ['#','~','G','H','C','I','D','-','F','I','N','I','S','H','-','4','~','#']
%STDOUT: #~GHCID-START~##~GHCID-START~#current working directory:
%STDOUT:   /home/jordan/ghcid-line-pragmas
%STDOUT: module import search paths:
%STDOUT:   .
%STDOUT: #~GHCID-START~##~GHCID-FINISH-4~#
%STDERR: #~GHCID-FINISH-4~#
%MESSAGES: [Loading {loadModule = "LinePragmaBad", loadFile = "/home/jordan/ghcid-line-pragmas/LinePragmaBad.hs"}]
%LOADED: ["/home/jordan/ghcid-line-pragmas/LinePragmaBad.hs"]

All good (1 module, at 16:49:27)

%WAITING: /home/jordan/ghcid-line-pragmas/.ghcid /home/jordan/ghcid-line-pragmas/LinePragmaBad.hs
%WAITING: /home/jordan/ghcid-line-pragmas
^C%STDIN: :quit
%STDIN: INTERNAL_GHCID.putStrLn ['#','~','G','H','C','I','D','-','F','I','N','I','S','H','-','5','~','#']
INTERNAL_GHCID.hPutStrLn INTERNAL_GHCID.stderr ['#','~','G','H','C','I','D','-','F','I','N','I','S','H','-','5','~','#']
%STDOUT: #~GHCID-START~##~GHCID-START~#Leaving GHCi.
```

**Comment by ndmitchell** [comment]
Thanks for the bug report. The problem is on https://github.com/ndmitchell/ghcid/blob/master/src/Language/Haskell/Ghcid/Parser.hs#L53 - it checks that the reported file has a specific extension. Given the existence of LINE pragmas perhaps that whole line just needs removing. Can you try removing it and seeing if it fixes the problem for you? If so, I'd welcome a PR (but happy to do it myself if you prefer).

That line dates back 5 years, and 3 years ago I added the `hs-boot` and `lhs-boot` cases to better support boot files. I imagine it's now just harmful!

**Comment by woehr** [comment]
I'm more than happy to give it a try! I didn't have the time to track it down myself, but knowing where to look, I'll give it a go.

**Comment by ndmitchell** [comment]
Closed in #276
